server/src/main/java/org/opensearch/gateway/remote/routingtable/IndexRoutingTableInput.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway.remote.routingtable;
+
+import org.opensearch.cluster.routing.IndexRoutingTable;
+import org.opensearch.cluster.routing.IndexShardRoutingTable;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.core.common.bytes.BytesReference;
+import org.opensearch.core.common.io.stream.BufferedChecksumStreamOutput;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+/**
+ * Class for creating object for uploading IndexRoutingTable to remote store.
+ */
+public class IndexRoutingTableInput {
```

Comment:
Can this extend `Writeable` interface

Reply:
Addressed in parent PR https://github.com/opensearch-project/OpenSearch/pull/13577/files#diff-385ee0bebd02caaf186ee5d8cf24d1e32aae519b682700f0f25941440876679cR27